---
layout: post
title: Mesophotic *Pachyseris speciosa* RNAseq data processing
date: '2022-12-20'
categories: Protocols
tags: [RNASeq, Bioinformatics]
---

# Mesophotic *Pachyseris speciosa* RNAseq data CBASS heat stress experiment

### Analyzed by Veronica Radice (Barshis Lab), Old Dominion University

## Background
*Field collection*

Fragments of mesophotic colonies of the reef-building coral *Pachyseris speciosa* (n=10, 35.7 to 37.6 m) were collected using closed circuit rebreathers at Leone on the island of Tutuila, American Sāmoa in the austral summer (-14.342, -170.789; February 28, 2022). Fragments of each coral were immediately preserved in the field in Zymo DNA/RNA Shield (TField) and the rest of the corals were transported live (~11 minute drive) to the experiment location. 

*Acute heat stress experiment*
An acute heat stress experiment using the CBASS (Coral Bleaching Automated Stress System) was conducted on the same day as the coral collection. Coral fragments were glued to pre-made genotype cards and placed in tanks at ambient temperature (29°C) while the fragments for the initial timepoint (T0) were dark-acclimated for ~30 minutes with an aerator prior to PAM measurements (DIVING PAM II; Walz, Effeltrich, Germany), n=3 measurements per fragment). One fragment of each genotype/species were placed in four tanks with different maximum target temperatures (Control=29°C, Low= 33°C, Medium= 35°C, High=38°C). Temperatures were selected relative to the local maximum monthly mean for American Sāmoa and increased from the control temperature for 3 hours until reaching the target temperature of each tank (profile maintained by Novatech Ice Probes and Bulk Reef Supply 100W Titanium Heaters). Temperature profiles for each tank were controlled via an Arduino with each temperature held (3 hours) followed by a ramp down back to control temperature (~1 hour). Water flow and turnover was maintained in each tank throughout the experiment (SUN JVP Series).

At the end of the maximum temperature hold (T2, after last daylight) 7 hours after the initial timepoint T0, fragments were sampled, photographed, and preserved in Zymo DNA/RNA Shield. The following morning (T3) 18 hours after the initial timepoint T0, PAM measurements were made on each coral fragment prior to being photographed and preserved in DNA/RNA Shield. All preserved coral fragments were frozen at -80°C until subsequent analysis.

--------------------------------------------------------------------------------------------

# Data
Raw data
> /RC/group/rc_barshis_lab/taxonarchive/2022-10-31_BMKData/BMK_DATA_20221021175354_1

Data I'm working on
> /cm/shared/courses/dbarshis/barshislab/VRad/taxons/2022-10-31_BMKData/

--------------------------------------------------------------------------------------------

## Summary

- RNA was extracted for n=50 *Pachyseris speciosa* holobiont samples from 5 individuals (genotypes 1, 3, 6, 8, 10)
- RNA was successfully sequenced for n=43 samples (paired end)

--------------------------------------------------------------------------------------------

### Rename sequencing files

Using /cm/shared/courses/dbarshis/21AdvGenomics/scripts/renamer_advbioinf.py

Already done.

--------------------------------------------------------------------------------------------

# Trimgalore

> /cm/shared/courses/dbarshis/barshislab/VRad/taxons/2022-10-31_BMKData/Data/

cp *_Pspe_* /cm/shared/courses/dbarshis/barshislab/VRad/taxons/Pachyseris_speciosa/data_RNAseq/

> /cm/shared/courses/dbarshis/barshislab/VRad/taxons/Pachyseris_speciosa/data_RNAseq/

```
nano trimgalore_Pspe.sh
```

```
#!/bin/bash -l

#SBATCH -o 2023-01-06_trimgalore_Pspe.txt
#SBATCH -n 1
#SBATCH --mail-user=vradice@odu.edu
#SBATCH --mail-type=END
#SBATCH --job-name=trimgalore_Pspe

enable_lmod

module load container_env trim_galore

for i in *Pspe*_R1.fq.gz ; do `crun trim_galore --fastqc --paired $i ${i%_R1.fq.gz}_R2.fq.gz`; done
```

```
sbatch trimgalore_Pspe.sh
```

Check job status
```
squeue -u vradice
```

Details:
Trim Galore version: 0.6.7
Cutadapt version: 3.4
Number of cores used for trimming: 1
Quality Phred score cutoff: 20
Quality encoding type selected: ASCII+33
Using Illumina adapter for trimming (count: 8467). Second best hit was Nextera (count: 4)
Adapter sequence: 'AGATCGGAAGAGC' (Illumina TruSeq, Sanger iPCR; auto-detected)
Maximum trimming error rate: 0.1 (default)
Minimum required adapter overlap (stringency): 1 bp
Minimum required sequence length for both reads before a sequence pair gets removed: 20 bp
Running FastQC on the data once trimming has completed
Output file will be GZIP compressed

This is cutadapt 3.4 with Python 3.9.7

--------------------------------------------------------------------------------------------

# MultiQC

MultiQC is a reporting tool that parses summary statistics from results and log files generated by other bioinformatics tools.
[https://multiqc.info/docs/#using-multiqc](https://multiqc.info/docs/#using-multiqc)

[https://hbctraining.github.io/Intro-to-rnaseq-fasrc-salmon-flipped/lessons/11_multiQC.html](https://hbctraining.github.io/Intro-to-rnaseq-fasrc-salmon-flipped/lessons/11_multiQC.html)


> /cm/shared/courses/dbarshis/barshislab/VRad/taxons/Pachyseris_speciosa/data_RNAseq/

```
enable_lmod
```

```
module load container_env multiqc
```

```
crun multiqc ./
```

Output:
```
  /// MultiQC 🔍 | v1.13

|           multiqc | MultiQC Version v1.14 now available!
|           multiqc | Search path : /cm/shared/courses/dbarshis/barshislab/VRad/taxons/Pachyseris_speciosa/data_RNAseq
|         searching | ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 100% 432/432  
|          cutadapt | Found 86 reports
|            fastqc | Found 86 reports
|           multiqc | Compressing plot data
|           multiqc | Report      : multiqc_report.html
|           multiqc | Data        : multiqc_data
|           multiqc | MultiQC complete
```

--------------------------------------------------------------------------------------------

# Symbiont typing

- based on internal transcribed spacer 2 (ITS2) region, a multicopy genetic marker commonly used to analyse Symbiodinium diversity
- map samples against ITS2 database
- Arif et al. 2014 [https://onlinelibrary.wiley.com/doi/full/10.1111/mec.12869](https://onlinelibrary.wiley.com/doi/full/10.1111/mec.12869)
- Corrigendum, Arif et al. 2019, with updated ITS2 database file [https://onlinelibrary.wiley.com/doi/10.1111/mec.14956](https://onlinelibrary.wiley.com/doi/10.1111/mec.14956)
- File:  mec14956-sup-0001-files1_corrigendum.fasta

***use Corrigendum Arif ITS2 2019 database***
> /cm/shared/courses/dbarshis/barshislab/VRad/taxons/Siderastrea_siderea/refassembly/
> /cm/shared/courses/dbarshis/barshislab/VRad/taxons/Siderastrea_siderea/symbiont_typing/

#### Check assembly details
```
/cm/shared/courses/dbarshis/18AdvBioinf/scripts/avg_cov_len_fasta_advbioinf.py mec14956-sup-0001-files1_corrigendum.fasta
```

```
The total number of sequences is 400
The average sequence length is 376
The total number of bases is 150400
The minimum sequence length is 376
The maximum sequence length is 376
The N50 is 376
Median Length = 376
contigs < 150bp = 0
contigs >= 500bp = 0
contigs >= 1000bp = 0
contigs >= 2000bp = 0
```

--------------------------------------------------------------------------------------------

## Map all samples to Arif ITS2 (all clades) 2019 database

> /cm/shared/courses/dbarshis/barshislab/VRad/taxons/Pachyseris_speciosa/data_RNAseq/

cp /cm/shared/courses/dbarshis/barshislab/VRad/taxons/Siderastrea_siderea/symbiont_typing/mec14956-sup-0001-files1_corrigendum.fasta /cm/shared/courses/dbarshis/barshislab/VRad/taxons/Pachyseris_speciosa/data_RNAseq/

##### Unzip .gz files
This will extract the compressed file and decompress it.
*Example*
gzip -d calc-sheet.ods.gz

To retain the archive file, we need to use the -k (keep) option again, as well as the -d (decompress) option.
*Example*
gzip -dk calc-sheet.ods.gz

Did not work - for some reason -k flag does not exist here
```
gzip -dk *_val_*.fq.gz
```
gzip: invalid option -- 'k'

```
gzip --help
```

*Actually do NOT need to unzip (later will need gzip files for STAR alignment)

```
nano gzip_Pspe.sh
```

```
#!/bin/bash -l

#SBATCH -n 1
#SBATCH --mail-user=vradice@odu.edu
#SBATCH --mail-type=ALL
#SBATCH --job-name=gzip_Pspe

gzip -d *_val_*.fq.gz
```

```
sbatch gzip_Pspe.sh
```

#### Build Bowtie index
Build a Bowtie index from a set of DNA sequences (creates 6 files for mapping).
The script bowtie2-build which is part of Bowtie2 requires Python3. 

python --version

python/3.6 (Default)

```
module avail bowtie2
```
bowtie2/2.4 (Default)
samtools/1.10.2

```
nano bowtie2-build.sh
```

```
#!/bin/bash -l

#SBATCH -o bowtie2-build.txt
#SBATCH -n 1
#SBATCH --mail-user=vradice@odu.edu
#SBATCH --mail-type=END
#SBATCH --job-name=bowtie2-build

enable_lmod
module load container_env perl
module load container_env python3
module load bowtie2/2.4

crun.bowtie2 bowtie2-build /cm/shared/courses/dbarshis/barshislab/VRad/taxons/Pachyseris_speciosa/data_RNAseq/mec14956-sup-0001-files1_corrigendum.fasta Arif_ITS2_corrigendum
```

```
sbatch bowtie2-build.sh
```

### Mapping to Arif_ITS2_corrigendum reference

> /cm/shared/courses/dbarshis/barshislab/VRad/taxons/Pachyseris_speciosa/data_RNAseq/

```
nano mapreads_Arif_ITS2_corrigendum_Pspe.sh
```

Files ending in 
_val_2.fq.gz
_val_1.fq.gz

So use *_val_*.fq

```
#!/bin/bash -l

#SBATCH -o bowtie2_Arif_ITS2_corrigendum_Pspe.txt
#SBATCH -n 1
#SBATCH --mail-user=vradice@odu.edu
#SBATCH --mail-type=END
#SBATCH --job-name=bowtie2_Arif_ITS2_corrigendum_Pspe

enable_lmod
module load container_env perl
module load container_env python3
module load bowtie2/2.4

for i in *_val_*.fq ; do bowtie2 --rg-id ${i%_*_val_*.fq} \
--rg SM:${i%*_val_*.fq} \
--local -x Arif_ITS2_corrigendum -U $i \
> ${i%*_val_*.fq}_Arif_ITS2_corrigendum.sam -k 5\n; done
```

```
sbatch mapreads_Arif_ITS2_corrigendum_Pspe.sh
```

--------------------------------------------------------------------------------------------

## Count expression - all reads mapped to Arif_ITS2 symbiont database reference

> /cm/shared/courses/dbarshis/barshislab/VRad/taxons/Pachyseris_speciosa/data_RNAseq/

```
nano countexpression_Arif_ITS2_corrigendum.sh
```

```
#!/bin/bash -l

#SBATCH -o countexpression_Arif_ITS2_corrigendum.txt
#SBATCH -n 1
#SBATCH --mail-user=vradice@odu.edu
#SBATCH --mail-type=END
#SBATCH --job-name=countexpression_Arif_ITS2_corrigendum_Pspe

enable_lmod
module load container_env python2

crun python2 /cm/shared/courses/dbarshis/barshislab/CCourtney/PoloRNASeq/scripts/countxpression_SB_advbioinf.py /cm/shared/courses/dbarshis/barshislab/VRad/taxons/Pachyseris_speciosa/data_RNAseq/*_Arif_ITS2_corrigendum.sam
```

```
sbatch countexpression_Arif_ITS2_corrigendum.sh
```

--------------------------------------------------------------------------------------------

## Merge all _Arif_ITS2_corrigendum_counts.txt files into one big table 

> /cm/shared/courses/dbarshis/barshislab/VRad/taxons/Pachyseris_speciosa/data_RNAseq/

- first need to add column to each .txt file with unique sample id
- so we can later identify which sample has which ITS2 sequences

Using regular expressions:

for f in file1 file2 file3; do sed -i "s/$/\t$f/" $f; done

- For each file in the list, this will use sed to append to the end of each line a tab and the filename
- Using the -i flag with sed to perform a replacement in-place, overwriting the file
- Perform a substitution with s/PATTERN/REPLACEMENT/. 
- In this example PATTERN is $, the end of the line, and REPLACEMENT is \t (= a TAB), 
- and $f is the filename, from the loop variable. 
- The s/// command is within double-quotes so that the shell can expand variables
- sed is most practical for pattern substitution and saving in-place. 


```
nano append_filename.sh
```

```
#!/bin/bash -l

#SBATCH -o append_filename.txt
#SBATCH -n 1
#SBATCH --mail-user=vradice@odu.edu
#SBATCH --mail-type=END
#SBATCH --job-name=append_filename

for f in *_Arif_ITS2_corrigendum_counts.txt; do sed -i "s/$/\t$f/" $f; done
```

```
sbatch append_filename.sh
```

#### Concatenate files

```
cat *_Arif_ITS2_corrigendum_counts.txt > merged_ITS2_counts_Arif-corrigendum_Pspe.txt
```

*Copied file to local folder on computer*
File:  merged_ITS2_counts_Arif-corrigendum_Pspe.xlsx

## Outcome symbiont clade mapping

- 91% reads mapped to I sequences
  - BLAST confirmed 100% match to former Symbiodinium sp. partial rRNA genes and ITS2
- Overall 95% of reads mapped to clade C
  - Not considering I sequences
- Need to test a few different Cladocopium references, pick reference with the best mapping rate

Bongaerts et al. (2021) Indo-Pacific *Pachyseris speciosa* mapped to *C. goreaui (C1)*

C3 is C. thermophilum 	C3-Gulf

** download Howells paper
https://onlinelibrary.wiley.com/doi/10.1111/mec.15372

https://www.frontiersin.org/articles/10.3389/fmicb.2019.02343/full includes a tree with different C3 types

--------------------------------------------------------------------------------------------

## Test symbiont reference transcriptome
### *Cladocopium thermophilum* (C3)

- Voolstra et al. unpublished reference transcriptome (private)
- File:  symb_arab_longest.fa

> /cm/shared/courses/dbarshis/barshislab/VRad/taxons/Pachyseris_speciosa/transcriptome_Sym/

## Check assembly details
```
/cm/shared/courses/dbarshis/18AdvBioinf/scripts/avg_cov_len_fasta_advbioinf.py symb_arab_longest.fa
```

```
The total number of sequences is 20428
The average sequence length is 528
The total number of bases is 10793533
The minimum sequence length is 201
The maximum sequence length is 8252
The N50 is 640
Median Length = 948
contigs < 150bp = 0
contigs >= 500bp = 6797
contigs >= 1000bp = 2071
contigs >= 2000bp = 351
```

#### Build Bowtie index
Build a Bowtie index from a set of DNA sequences (creates 6 files for mapping).

```
enable_lmod
module load container_env perl
module load container_env python3
module load bowtie2/2.4

crun.bowtie2 bowtie2-build /cm/shared/courses/dbarshis/barshislab/VRad/taxons/Pachyseris_speciosa/transcriptome_Sym/symb_arab_longest.fa sym_C-thermophilum
```

### Mapping to symbiont *Cladocopium thermophilum* transcriptome reference

> /cm/shared/courses/dbarshis/barshislab/VRad/taxons/Pachyseris_speciosa/data_RNAseq/

```
nano mapreads_Cladocopium-thermophilum.sh
```

```
#!/bin/bash -l

#SBATCH -o bowtie2_Cladocopium-thermophilum.txt
#SBATCH -n 1
#SBATCH --mail-user=vradice@odu.edu
#SBATCH --mail-type=END
#SBATCH --job-name=bowtie2_Cladocopium-thermophilum

enable_lmod
module load container_env perl
module load container_env python3
module load bowtie2/2.4

for i in *_val_*.fq ; do bowtie2 --rg-id ${i%_*_val_*.fq} \
--rg SM:${i%*_val_*.fq} \
--local -x sym_C-thermophilum -U $i \
> ${i%*_val_*.fq}_sym_C-thermophilum.sam -k 5\n; done
```

```
sbatch mapreads_Cladocopium-thermophilum.sh
```

--------------------------------------------------------------------------------------------

## Count expression - all reads mapped to symbiont *Cladocopium thermophilum* transcriptome reference

> /cm/shared/courses/dbarshis/barshislab/VRad/taxons/Pachyseris_speciosa/data_RNAseq/

```
nano countexpression_sym_C-thermophilum.sh
```

```
#!/bin/bash -l

#SBATCH -o countexpression_sym_C-thermophilum.txt
#SBATCH -n 1
#SBATCH --mail-user=vradice@odu.edu
#SBATCH --mail-type=END
#SBATCH --job-name=countexpression_sym_C-thermophilum

enable_lmod
module load container_env python2

crun python2 /cm/shared/courses/dbarshis/barshislab/CCourtney/PoloRNASeq/scripts/countxpression_SB_advbioinf.py /cm/shared/courses/dbarshis/barshislab/VRad/taxons/Pachyseris_speciosa/data_RNAseq/*_sym_C-thermophilum.sam
```

```
sbatch countexpression_sym_C-thermophilum.sh
```

Outcome:
- avg. 3.53% singly aligned (range 2.14-8.19%)
- avg. 3.98% overall alignment (range 2.39-8.75%)

--------------------------------------------------------------------------------------------

## Test symbiont reference transcriptome
### *Cladocopium goreaui* (C1) Davies reference

- Davies et al. (2018). *Symbiodinium functional diversity in the coral Siderastrea siderea is influenced by thermal stress and reef environment, but not ocean acidification.* Frontiers in Marine Science. FMARS-05-00150.
- Assembled and annotated transcriptome for the symbiotic dinoflagellate algae Cladocopium hosted by *Siderastrea siderea* with all host contamination removed. 
- Data were generated using Illumina HiSeq2000 2*100bp reads and assembled using Trinity.
- File:  davies_cladeC_feb.fasta
[https://sites.bu.edu/davieslab/data-code/](https://sites.bu.edu/davieslab/data-code/)
[https://doi.org/10.3389/fmars.2018.00150](https://doi.org/10.3389/fmars.2018.00150)

> /cm/shared/courses/dbarshis/barshislab/VRad/taxons/Siderastrea_siderea/refassembly/

## Check assembly details
```
/cm/shared/courses/dbarshis/18AdvBioinf/scripts/avg_cov_len_fasta_advbioinf.py 65838_davies_cladeC_feb_fixed-clean_suffixed.fasta
```

```
The total number of sequences is 65838
The average sequence length is 1482
The total number of bases is 97581498
The minimum sequence length is 500
The maximum sequence length is 18168
The N50 is 1746
Median Length = 979
contigs < 150bp = 0
contigs >= 500bp = 65838
contigs >= 1000bp = 40840
contigs >= 2000bp = 13246
```

#### Build Bowtie index
Build a Bowtie index from a set of DNA sequences (creates 6 files for mapping).

```
enable_lmod
module load container_env perl
module load container_env python3
module load bowtie2/2.4

crun.bowtie2 bowtie2-build /cm/shared/courses/dbarshis/barshislab/VRad/taxons/Siderastrea_siderea/refassembly/65838_davies_cladeC_feb_fixed-clean_suffixed.fasta sym_C-goreaui_Davies
```

### Mapping to symbiont *Cladocopium goreaui* Davies transcriptome reference

> /cm/shared/courses/dbarshis/barshislab/VRad/taxons/Pachyseris_speciosa/data_RNAseq/

```
nano mapreads_Cladocopium-goreaui_Davies.sh
```

```
#!/bin/bash -l

#SBATCH -o bowtie2_Cladocopium-goreaui_Davies.txt
#SBATCH -n 1
#SBATCH --mail-user=vradice@odu.edu
#SBATCH --mail-type=END
#SBATCH --job-name=bowtie2_Cladocopium-goreaui_Davies

enable_lmod
module load container_env perl
module load container_env python3
module load bowtie2/2.4

for i in *_val_*.fq ; do bowtie2 --rg-id ${i%_*_val_*.fq} \
--rg SM:${i%*_val_*.fq} \
--local -x sym_C-goreaui_Davies -U $i \
> ${i%*_val_*.fq}_sym_C-goreaui_Davies.sam -k 5\n; done
```

```
sbatch mapreads_Cladocopium-goreaui_Davies.sh
```

--------------------------------------------------------------------------------------------

## Count expression - all reads mapped to symbiont *Cladocopium goreaui* Davies transcriptome reference

> /cm/shared/courses/dbarshis/barshislab/VRad/taxons/Pachyseris_speciosa/data_RNAseq/

```
nano countexpression_sym_C-goreaui_Davies.sh
```

```
#!/bin/bash -l

#SBATCH -o countexpression_sym_C-goreaui_Davies.txt
#SBATCH -n 1
#SBATCH --mail-user=vradice@odu.edu
#SBATCH --mail-type=END
#SBATCH --job-name=countexpression_sym_C-goreaui_Davies

enable_lmod
module load container_env python2

crun python2 /cm/shared/courses/dbarshis/barshislab/CCourtney/PoloRNASeq/scripts/countxpression_SB_advbioinf.py /cm/shared/courses/dbarshis/barshislab/VRad/taxons/Pachyseris_speciosa/data_RNAseq/*_sym_C-goreaui_Davies.sam
```

```
sbatch countexpression_sym_C-goreaui_Davies.sh
```

Outcome:
- avg. 3.53% singly aligned (range 1.05-18.96%)
- avg. 5.37% overall alignment (range 1.73-27.10%)

--------------------------------------------------------------------------------------------

## Test symbiont reference transcriptome 
### *Cladocopium goreaui* (C1) Chen reference 

- Chen et al. (2020). Evidence That Inconsistent Gene Prediction Can Mislead Analysis of Dinoflagellate Genomes. Journal of Phycology.
- predicted protein‐coding genes from published draft Symbiodiniaceae genome of *Cladocopium goreaui* from Liu et al. (2018)
    - Liu et al. (2018). Symbiodinium genomes reveal adaptive evolution of functions related to coral‐dinoflagellate symbiosis. Commun. Biol. 1:95.
        - Symbiodinium goreaui (Clade C, type C1; AIMS-aten-C1-MI-cfu-B2, now AIMS culture collection SCF055-01) is a single-cell monoclonal culture first isolated from the coral Acropora tenuis at Magnetic Island (Queensland, Australia) at 3 m depth; this culture is maintained at the Australian Institute of Marine Science, Townsville, Australia.
        - Illumina HiSeq 2500 platform, 2 × 150 bp reads
    - generated 116.0 Gb (614.6 million reads)
- file:  Cladocopium_goreaui.CDS.fna
[https://onlinelibrary.wiley.com/doi/full/10.1111/jpy.12947](https://onlinelibrary.wiley.com/doi/full/10.1111/jpy.12947)
- Chen et al. (2019). Revised genome sequences and annotations of six Symbiodiniaceae taxa.
[https://espace.library.uq.edu.au/view/UQ:8279c9a](https://espace.library.uq.edu.au/view/UQ:8279c9a)

> /cm/shared/courses/dbarshis/barshislab/VRad/taxons/Siderastrea_siderea/refassembly/

## Check assembly details
```
/cm/shared/courses/dbarshis/18AdvBioinf/scripts/avg_cov_len_fasta_advbioinf.py 39006_Cladocopium_goreaui.CDS.fna
```

```
The total number of sequences is 39006
The average sequence length is 1625
The total number of bases is 63419290
The minimum sequence length is 111
The maximum sequence length is 40317
The N50 is 2388
Median Length = 852
contigs < 150bp = 5
contigs >= 500bp = 31703
contigs >= 1000bp = 21315
contigs >= 2000bp = 9624
```

#### Build Bowtie index
Build a Bowtie index from a set of DNA sequences (creates 6 files for mapping).

```
enable_lmod
module load container_env perl
module load container_env python3
module load bowtie2/2.4

crun.bowtie2 bowtie2-build /cm/shared/courses/dbarshis/barshislab/VRad/taxons/Siderastrea_siderea/refassembly/39006_Cladocopium_goreaui.CDS.fna sym_C-goreaui_Chen
```

### Mapping to symbiont *Cladocopium goreaui* Chen transcriptome reference

> /cm/shared/courses/dbarshis/barshislab/VRad/taxons/Pachyseris_speciosa/data_RNAseq/

```
nano mapreads_Cladocopium-goreaui_Chen.sh
```

```
#!/bin/bash -l

#SBATCH -o bowtie2_Cladocopium-goreaui_Chen.txt
#SBATCH -n 1
#SBATCH --mail-user=vradice@odu.edu
#SBATCH --mail-type=END
#SBATCH --job-name=bowtie2_Cladocopium-goreaui_Chen

enable_lmod
module load container_env perl
module load container_env python3
module load bowtie2/2.4

for i in *_val_*.fq ; do bowtie2 --rg-id ${i%_*_val_*.fq} \
--rg SM:${i%*_val_*.fq} \
--local -x sym_C-goreaui_Chen -U $i \
> ${i%*_val_*.fq}_sym_C-goreaui_Chen.sam -k 5\n; done
```

```
sbatch mapreads_Cladocopium-goreaui_Chen.sh
```

--------------------------------------------------------------------------------------------

## Count expression - all reads mapped to symbiont *Cladocopium goreaui* Chen transcriptome reference

> /cm/shared/courses/dbarshis/barshislab/VRad/taxons/Pachyseris_speciosa/data_RNAseq/

```
nano countexpression_sym_C-goreaui_Chen.sh
```

```
#!/bin/bash -l

#SBATCH -o countexpression_sym_C-goreaui_Chen.txt
#SBATCH -n 1
#SBATCH --mail-user=vradice@odu.edu
#SBATCH --mail-type=END
#SBATCH --job-name=countexpressionsym_C-goreaui_Chen

enable_lmod
module load container_env python2

crun python2 /cm/shared/courses/dbarshis/barshislab/CCourtney/PoloRNASeq/scripts/countxpression_SB_advbioinf.py /cm/shared/courses/dbarshis/barshislab/VRad/taxons/Pachyseris_speciosa/data_RNAseq/*_sym_C-goreaui_Chen.sam
```

```
sbatch countexpression_sym_C-goreaui_Chen.sh
```

Outcome:
- avg. 3.77% singly aligned (range 1.06-20.05%)
- avg. 4.98% overall alignment (range 1.60-24.70%)

--------------------------------------------------------------------------------------------

## Decision about symbiont reference 

> /cm/shared/courses/dbarshis/barshislab/VRad/taxons/Siderastrea_siderea/refassembly/39006_Cladocopium_goreaui.CDS.fna

***Use Cladocopium goreaui Chen transcriptome**

#### Add suffix to reference
```
nano addsuffixtofastaseqnames.sh
```

```
#!/bin/bash -l

#SBATCH -o addsuffixtofastaseqnames.txt
#SBATCH -n 1
#SBATCH --mail-user=vradice@odu.edu
#SBATCH --mail-type=ALL
#SBATCH --job-name=addsuffixtofastaseqnames

/cm/shared/courses/dbarshis/barshislab/VRad/taxons/Porites_astreoides/Paytan/P_ast/refassembly/P_ast_Kenkel/addsuffixtofastaseqnames.py Chen_Symbiont /cm/shared/courses/dbarshis/barshislab/VRad/taxons/Siderastrea_siderea/refassembly/39006_Cladocopium_goreaui.CDS.fna
```

```
sbatch addsuffixtofastaseqnames.sh
```

output:
> /cm/shared/courses/dbarshis/barshislab/VRad/taxons/Siderastrea_siderea/refassembly/39006_Cladocopium_goreaui.C_suffixed.fasta

Rename fasta file based on # of contigs
```
mv 39006_Cladocopium_goreaui.C_suffixed.fasta 39006_Cladocopium_goreaui.Chen_suffixed.fasta
```

--------------------------------------------------------------------------------------------

## Coral Host *Pachyseris speciosa* genome (Bongaerts)

[Published genome: Bongaerts et al. (2021)](https://www.cell.com/current-biology/fulltext/S0960-9822(21)00367-5)
[NCBI project PRJNA686157](https://www.ncbi.nlm.nih.gov/bioproject/PRJNA686157)

> /cm/shared/courses/dbarshis/barshislab/VRad/taxons/Pachyseris_speciosa/genome_Bongaerts/

```
gzip -d pspe_final_0.12.fasta.gz
```

## Check assembly details
```
/cm/shared/courses/dbarshis/18AdvBioinf/scripts/avg_cov_len_fasta_advbioinf.py pspe_final_0.12.fasta
```

```
The total number of sequences is 2368
The average sequence length is 415722
The total number of bases is 984430632
The minimum sequence length is 13226
The maximum sequence length is 4615051
The N50 is 766649
Median Length = 234970
contigs < 150bp = 0
contigs >= 500bp = 2368
contigs >= 1000bp = 2368
contigs >= 2000bp = 2368
```

#### Add suffix to host reference
```
nano addsuffixtofastaseqnames.sh
```

```
#!/bin/bash -l

#SBATCH -o addsuffixtofastaseqnames.txt
#SBATCH -n 1
#SBATCH --mail-user=vradice@odu.edu
#SBATCH --mail-type=ALL
#SBATCH --job-name=addsuffixtofastaseqnames

/cm/shared/courses/dbarshis/barshislab/VRad/taxons/Porites_astreoides/Paytan/P_ast/refassembly/P_ast_Kenkel/addsuffixtofastaseqnames.py Pspeciosa_Bongaerts /cm/shared/courses/dbarshis/barshislab/VRad/taxons/Pachyseris_speciosa/genome_Bongaerts/pspe_final_0.12.fasta
```

```
sbatch addsuffixtofastaseqnames.sh
```

output:
> /cm/shared/courses/dbarshis/barshislab/VRad/taxons/Pachyseris_speciosa/genome_Bongaerts/pspe_final_0.12_suffixed.fasta

Rename fasta file based on # of contigs
```
mv pspe_final_0.12_suffixed.fasta 2368_pspe_final_0.12_suffixed.fasta
```

--------------------------------------------------------------------------------------------

## Concatenate host and symbiont references to make Holobiont reference

Host
> /cm/shared/courses/dbarshis/barshislab/VRad/taxons/Pachyseris_speciosa/genome_Bongaerts/2368_pspe_final_0.12_suffixed.fasta

Symbiont
> /cm/shared/courses/dbarshis/barshislab/VRad/taxons/Siderastrea_siderea/refassembly/39006_Cladocopium_goreaui.Chen_suffixed.fasta

```
cat /cm/shared/courses/dbarshis/barshislab/VRad/taxons/Pachyseris_speciosa/genome_Bongaerts/2368_pspe_final_0.12_suffixed.fasta /cm/shared/courses/dbarshis/barshislab/VRad/taxons/Siderastrea_siderea/refassembly/39006_Cladocopium_goreaui.Chen_suffixed.fasta > Pspe_Cgoreaui_hybridref.fasta
```

## Check assembly details

```
/cm/shared/courses/dbarshis/18AdvBioinf/scripts/avg_cov_len_fasta_advbioinf.py Pspe_Cgoreaui_hybridref.fasta
```

```
The total number of sequences is 41374
The average sequence length is 25326
The total number of bases is 1047849922
The minimum sequence length is 111
The maximum sequence length is 4615051
The N50 is 726185
Median Length = 1068
contigs < 150bp = 5
contigs >= 500bp = 34071
contigs >= 1000bp = 23683
contigs >= 2000bp = 11992
```

--------------------------------------------------------------------------------------------

#### Create a genome index for STAR

> /cm/shared/courses/dbarshis/barshislab/VRad/taxons/Pachyseris_speciosa/data_RNAseq/reference/

```
nano GenomeGenerate.sh
```

```
#!/bin/bash -l

#SBATCH -o 2023-01-23_Pspe_genomeGenerate.txt
#SBATCH -n 4
#SBATCH -N 1
#SBATCH --mail-user=vradice@odu.edu
#SBATCH --mail-type=END
#SBATCH --job-name=Pspegenomegenerate

enable_lmod
module load container_env star

STAR --runMode genomeGenerate --runThreadN 4 --genomeDir ./ --genomeFastaFiles /cm/shared/courses/dbarshis/barshislab/VRad/taxons/Pachyseris_speciosa/data_RNAseq/Pspe_Cgoreaui_hybridref.fasta --genomeChrBinNbits 16
```

```
sbatch GenomeGenerate.sh
```

--------------------------------------------------------------------------------------------

# Mapping - STAR Alignment (Spliced Transcripts Alignment to a Reference)
## Map all the samples to concatenated (host plus symbiont) hybrid reference

STAR is an aligner designed to specifically address many of the challenges of RNA-seq data mapping using a strategy to account for spliced alignments.
[https://hbctraining.github.io/Intro-to-rnaseq-hpc-O2/lessons/03_alignment.html#:~:text=STAR%20Alignment%20Strategy,Clustering%2C%20stitching%2C%20and%20scoring](https://hbctraining.github.io/Intro-to-rnaseq-hpc-O2/lessons/03_alignment.html#:~:text=STAR%20Alignment%20Strategy,Clustering%2C%20stitching%2C%20and%20scoring)

*Requires gzip format*
```
nano gzip_Pspe_2.sh
```

```
#!/bin/bash -l

#SBATCH -n 1
#SBATCH --mail-user=vradice@odu.edu
#SBATCH --mail-type=ALL
#SBATCH --job-name=gzip_Pspe

gzip *_val_*.fq
```

```
sbatch gzip_Pspe_2.sh
```

```
STAR --version
```
2.7.3a

I removed --outReadsUnmapped Fastx from script below. 

```
nano star_Pspe.sh
```

```
#!/bin/bash -l

#SBATCH -o 2023-01-26_PspeSTARMapping.txt
#SBATCH -n 16
#SBATCH -N 1
#SBATCH --mail-user=vradice@odu.edu
#SBATCH --mail-type=ALL
#SBATCH --job-name=STARPspe

enable_lmod
module load container_env star

for i in *_R1_val_1.fq.gz ; do `STAR --genomeDir /cm/shared/courses/dbarshis/barshislab/VRad/taxons/Pachyseris_speciosa/data_RNAseq/reference/ --runThreadN 16 --outSAMattributes All --outSAMattrRGline ID:${i%_R1_val_1.fq.gz} --genomeLoad LoadAndRemove --outFilterType Normal --outFilterMismatchNoverLmax 0.03 --outSAMstrandField intronMotif --outFilterIntronMotifs RemoveNoncanonical --outSAMtype BAM Unsorted --limitBAMsortRAM 5784458574 --readFilesCommand zcat --outFilterMatchNminOverLread 0.2 --outFilterScoreMinOverLread 0.2 --readFilesIn $i ${i%_R1_val_1.fq.gz}_R2_val_2.fq.gz --outFileNamePrefix ${i%_R1_val_1.fq.gz}_Pspe_Cgoreaui`; done
```

```
sbatch star_Pspe.sh
```

Failed again- ExitCode 127
Found two spaces in the script - could this cause the issue? Deleted the extra space
.bam files are complete, and MultiQC showed that the samples aligned to the holobiont reference.

```
enable_lmod
module load container_env multiqc
crun multiqc ./
```

```
  /// MultiQC 🔍 | v1.13

|           multiqc | MultiQC Version v1.14 now available!
|           multiqc | Search path : /cm/shared/courses/dbarshis/barshislab/VRad/taxons/Pachyseris_speciosa/data_RNAseq
|         searching | ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 100% 1401/1401  
|              star | Found 43 reports
|           bowtie2 | Found 4 reports
|          cutadapt | Found 86 reports
|            fastqc | Found 86 reports
|           multiqc | Compressing plot data
|           multiqc | Previous MultiQC output found! Adjusting filenames..
|           multiqc | Use -f or --force to overwrite existing reports instead
|           multiqc | Report      : multiqc_report_1.html
|           multiqc | Data        : multiqc_data_1
|           multiqc | MultiQC complete
```

Output:  multiqc_report_Pspe_Cgoreaui.html

Move *Aligned.out.bam files into new directory.

```
mkdir aligned_hybridref
mv *Aligned.out.bam aligned_hybridref/
cd aligned_hybridref
```

> /cm/shared/courses/dbarshis/barshislab/VRad/taxons/Pachyseris_speciosa/data_RNAseq/aligned_hybridref/

--------------------------------------------------------------------------------------------

## De-duplicate reads
### Use STAR to mark duplicates

*Deduplicate BEFORE salmon quantification.*

Serdar Turkarslan (ISB) tested two methods of deduplication and ended up using the one that is based on the STAR to keep it consistent with the rest of the pipeline. Here is the one we are implementing (STAR-samtools-based).

## Mark duplicates on STAR aligned BAM file

> /cm/shared/courses/dbarshis/barshislab/VRad/taxons/Pachyseris_speciosa/data_RNAseq/aligned_hybridref/

```
nano dedup_STAR.sh
```

```
#!/bin/bash -l

#SBATCH -o 2023-01-28_PspeSTARdeduplicate.txt
#SBATCH -n 16
#SBATCH -N 1
#SBATCH --mail-user=vradice@odu.edu
#SBATCH --mail-type=ALL
#SBATCH --job-name=STARPspe

enable_lmod
module load container_env star

STAR --runThreadN 32 --runMode inputAlignmentsFromBAM --bamRemoveDuplicatesType UniqueIdenticalNotMulti --inputBAMfile *Aligned.out.bam --outFileNamePrefix Dedup_Pspe
```

```
sbatch dedup_STAR.sh
```

STAR version=2.7.3a

## Remove singletons from marked STAR BAM file

*Best to run as sbatch.*

```
enable_lmod
module load container_env samtools

crun samtools view -@ 8 -b -F 0x400 Dedup_PspeProcessed.out.bam > Processed.out.noDup.bam
```

samtools/1.10.2 (Loaded,Default)

4.8G Jan 28 20:48 Dedup_PspeProcessed.out.bam
4.2G Jan 28 21:08 Processed.out.noDup.bam

## Collate reads by name and sort

```
nano collate_noDup.sh
```

```
#!/bin/bash -l

#SBATCH -o 2023-01-30_collate_noDup.txt
#SBATCH -n 8
#SBATCH -N 1
#SBATCH --mail-user=vradice@odu.edu
#SBATCH --mail-type=ALL
#SBATCH --job-name=collate

enable_lmod
module load container_env samtools

crun samtools sort -o Collated.noDup.bam -n -@ 8 Processed.out.noDup.bam
```

```
sbatch collate_noDup.sh
```

#### Samtools-based deduplication:
*Alternative to above*
(I did not use)

##### Add MS and MC tags for markdup to use later:

fixmate_cmd = 'samtools fixmate -m %s %s' %(aligned_bam,fixmate_bam)

##### position order

sort_cmd = 'samtools sort -o %s %s' %(ordered_bam,fixmate_bam)

##### mark duplicates

markdup_cmd = 'samtools markdup -r -s %s %s' %(ordered_bam,markdup_bam)

##### remove singletons

rmsingletons_cmd = 'samtools view -@ 8 -F 0x08 -b %s > %s' %(markdup_bam,nosingleton_bam)

--------------------------------------------------------------------------------------------

# SALMON quantification
## Count expression - generate counts files
### Take 1 (testing for first time)

*See next section for how it was actually run.*

> /cm/shared/courses/dbarshis/barshislab/VRad/taxons/Pachyseris_speciosa/data_RNAseq/aligned_hybridref/

```
enable_lmod
module load container_env salmon
module avail salmon
```
salmon/1.9.0

```
nano salmon_quant_dedup.sh
```

```
#!/bin/bash -l

#SBATCH -o 2023-01-30_salmon_quant_dedup.txt
#SBATCH -n 16
#SBATCH -N 1
#SBATCH --mail-user=vradice@odu.edu
#SBATCH --mail-type=ALL
#SBATCH --job-name=salmon_quant

enable_lmod
module load container_env salmon/1.9.0

crun salmon quant -t /cm/shared/courses/dbarshis/barshislab/VRad/taxons/Pachyseris_speciosa/data_RNAseq/Pspe_Cgoreaui_hybridref.fasta --libType A -a Collated.noDup.bam -o salmon_quant
```

```
sbatch salmon_quant_dedup.sh
```

> /cm/shared/courses/dbarshis/barshislab/VRad/taxons/Pachyseris_speciosa/data_RNAseq/aligned_hybridref/salmon_quant/

Output: quant.sf

*Problem that all individuals merged into one file, no individual sample-level data*

CONTINUE HERE



? Delete below ?
Earlier attempts to run salmon_quant BEFORE dedup bam files (all failed)

2023-01-27_salmon_quant.txt
Run time 00:09:10, FAILED, ExitCode 139

processed 44000000 reads in current round/var/spool/slurmd/job9826032/slurm_script: line 13: 95738 Segmentation fault

Exit Code 139 is also known as a Segmentation Fault, commonly referenced by the shorthand SIGSEGV. Typically, this fault occurs when a Kubernetes container attempts to access a memory repository that it doesn’t have permission to access. Due to this attempt at entering a repository that is not assigned to the particular container, it will then be terminated.

```
nano salmon_quant_b.sh
```

```
#!/bin/bash -l

#SBATCH -o 2023-01-27_salmon_quant_b.txt
#SBATCH -n 16
#SBATCH -N 1
#SBATCH --mail-user=vradice@odu.edu
#SBATCH --mail-type=ALL
#SBATCH --job-name=salmon_quant

enable_lmod
module load container_env salmon/1.9.0

for i in *Aligned.out.bam ; do `crun salmon quant -t /cm/shared/courses/dbarshis/barshislab/VRad/taxons/Pachyseris_speciosa/data_RNAseq/Pspe_Cgoreaui_hybridref.fasta --libType A -a *Aligned.out.bam -o salmon_quant`; done
```

```
sbatch salmon_quant_b.sh
```

2023-01-27_salmon_quant_b.txt
Run time 05:48:40, FAILED, ExitCode 139

Empty file /cm/shared/courses/dbarshis/barshislab/VRad/taxons/Pachyseris_speciosa/data_RNAseq/aligned_hybridref/salmon_quant/logs/salmon_quant.log

/cm/shared/courses/dbarshis/barshislab/VRad/taxons/Pachyseris_speciosa/data_RNAseq/aligned_hybridref/
cat 2023-01-27_salmon_quant_b.txt

^ Can't tell what error is, looks like it went through to the last file ?


Katie ran multiple batches with multiple files

--------------------------------------------------------------------------------------------

# Combined STAR deduplication and SALMON quantification
## Count expression - generate counts files
### Final

> /cm/shared/courses/dbarshis/barshislab/VRad/taxons/Pachyseris_speciosa/data_RNAseq/aligned_hybridref/

```
enable_lmod
module load container_env salmon
module avail salmon
```
salmon/1.9.0

From Dan: generate a line by line bash shell script with one line for each file so you can specify the -o directory naming. 
Basically, you'll have your sbatch header and then a line for each .bam file for deduping and salmon quanting

ODU HPC cluster:
*All of the containers in our environment follow a naming scheme where crun will just run them, but crun.<program name> will explicitly also run that program’s container if you have loaded the associated module.  This is to help people with workflows that involve multiple containers, because if I module load several times, crun would only let me run the last loaded module.*

But script failed because "crun.STAR: command not found".

NB: Script below does not include 20220228_Meso_Pspe_10_C_T2_Pspe_CgoreauiAligned.out.bam because already ran this file as a test.

```
nano STAR-dedup_salmon-quant.sh
```

```
#!/bin/bash -l

#SBATCH -o 2023-02-01_STAR-dedup_salmon-quant.txt
#SBATCH -n 16
#SBATCH -N 1
#SBATCH --mail-user=vradice@odu.edu
#SBATCH --mail-type=ALL
#SBATCH --job-name=salmon_quant

enable_lmod
module load container_env salmon/1.9.0
module load container_env star

STAR --runThreadN 32 --runMode inputAlignmentsFromBAM --bamRemoveDuplicatesType UniqueIdenticalNotMulti --inputBAMfile 20220228_Meso_Pspe_10_C_T3_Pspe_CgoreauiAligned.out.bam --outFileNamePrefix 20220228_Meso_Pspe_10_C_T3_Pspe_CgoreauiAligned_deduped.bamProcessed.out.bam

crun.salmon quant -t /cm/shared/courses/dbarshis/barshislab/VRad/taxons/Pachyseris_speciosa/data_RNAseq/Pspe_Cgoreaui_hybridref.fasta --libType A -a 20220228_Meso_Pspe_10_C_T3_Pspe_CgoreauiAligned_deduped.bamProcessed.out.bam -o 20220228_Meso_Pspe_10_C_T3_Pspe_CgoreauiAligned_deduped_salmon_quant

STAR --runThreadN 32 --runMode inputAlignmentsFromBAM --bamRemoveDuplicatesType UniqueIdenticalNotMulti --inputBAMfile 20220228_Meso_Pspe_10_H_T2_Pspe_CgoreauiAligned.out.bam --outFileNamePrefix 20220228_Meso_Pspe_10_H_T2_Pspe_CgoreauiAligned_deduped.bamProcessed.out.bam

crun.salmon quant -t /cm/shared/courses/dbarshis/barshislab/VRad/taxons/Pachyseris_speciosa/data_RNAseq/Pspe_Cgoreaui_hybridref.fasta --libType A -a 20220228_Meso_Pspe_10_H_T2_Pspe_CgoreauiAligned_deduped.bamProcessed.out.bam -o 20220228_Meso_Pspe_10_H_T2_Pspe_CgoreauiAligned_deduped_salmon_quant

STAR --runThreadN 32 --runMode inputAlignmentsFromBAM --bamRemoveDuplicatesType UniqueIdenticalNotMulti --inputBAMfile 20220228_Meso_Pspe_10_L_T2_Pspe_CgoreauiAligned.out.bam --outFileNamePrefix 20220228_Meso_Pspe_10_L_T2_Pspe_CgoreauiAligned_deduped.bamProcessed.out.bam

crun.salmon quant -t /cm/shared/courses/dbarshis/barshislab/VRad/taxons/Pachyseris_speciosa/data_RNAseq/Pspe_Cgoreaui_hybridref.fasta --libType A -a 20220228_Meso_Pspe_10_L_T2_Pspe_CgoreauiAligned_deduped.bamProcessed.out.bam -o 20220228_Meso_Pspe_10_L_T2_Pspe_CgoreauiAligned_deduped_salmon_quant

STAR --runThreadN 32 --runMode inputAlignmentsFromBAM --bamRemoveDuplicatesType UniqueIdenticalNotMulti --inputBAMfile 20220228_Meso_Pspe_10_L_T3_Pspe_CgoreauiAligned.out.bam --outFileNamePrefix 20220228_Meso_Pspe_10_L_T3_Pspe_CgoreauiAligned_deduped.bamProcessed.out.bam

crun.salmon quant -t /cm/shared/courses/dbarshis/barshislab/VRad/taxons/Pachyseris_speciosa/data_RNAseq/Pspe_Cgoreaui_hybridref.fasta --libType A -a 20220228_Meso_Pspe_10_L_T3_Pspe_CgoreauiAligned_deduped.bamProcessed.out.bam -o 20220228_Meso_Pspe_10_L_T3_Pspe_CgoreauiAligned_deduped_salmon_quant

STAR --runThreadN 32 --runMode inputAlignmentsFromBAM --bamRemoveDuplicatesType UniqueIdenticalNotMulti --inputBAMfile 20220228_Meso_Pspe_10_M_T2_Pspe_CgoreauiAligned.out.bam --outFileNamePrefix 20220228_Meso_Pspe_10_M_T2_Pspe_CgoreauiAligned_deduped.bamProcessed.out.bam

crun.salmon quant -t /cm/shared/courses/dbarshis/barshislab/VRad/taxons/Pachyseris_speciosa/data_RNAseq/Pspe_Cgoreaui_hybridref.fasta --libType A -a 20220228_Meso_Pspe_10_M_T2_Pspe_CgoreauiAligned_deduped.bamProcessed.out.bam -o 20220228_Meso_Pspe_10_M_T2_Pspe_CgoreauiAligned_deduped_salmon_quant

STAR --runThreadN 32 --runMode inputAlignmentsFromBAM --bamRemoveDuplicatesType UniqueIdenticalNotMulti --inputBAMfile 20220228_Meso_Pspe_10_M_T3_Pspe_CgoreauiAligned.out.bam --outFileNamePrefix 20220228_Meso_Pspe_10_M_T3_Pspe_CgoreauiAligned_deduped.bamProcessed.out.bam

crun.salmon quant -t /cm/shared/courses/dbarshis/barshislab/VRad/taxons/Pachyseris_speciosa/data_RNAseq/Pspe_Cgoreaui_hybridref.fasta --libType A -a 20220228_Meso_Pspe_10_M_T3_Pspe_CgoreauiAligned_deduped.bamProcessed.out.bam -o 20220228_Meso_Pspe_10_M_T3_Pspe_CgoreauiAligned_deduped_salmon_quant

STAR --runThreadN 32 --runMode inputAlignmentsFromBAM --bamRemoveDuplicatesType UniqueIdenticalNotMulti --inputBAMfile 20220228_Meso_Pspe_10_T0_Pspe_CgoreauiAligned.out.bam --outFileNamePrefix 20220228_Meso_Pspe_10_T0_Pspe_CgoreauiAligned_deduped.bamProcessed.out.bam

crun.salmon quant -t /cm/shared/courses/dbarshis/barshislab/VRad/taxons/Pachyseris_speciosa/data_RNAseq/Pspe_Cgoreaui_hybridref.fasta --libType A -a 20220228_Meso_Pspe_10_T0_Pspe_CgoreauiAligned_deduped.bamProcessed.out.bam -o 20220228_Meso_Pspe_10_T0_Pspe_CgoreauiAligned_deduped_salmon_quant

STAR --runThreadN 32 --runMode inputAlignmentsFromBAM --bamRemoveDuplicatesType UniqueIdenticalNotMulti --inputBAMfile 20220228_Meso_Pspe_10_TF_Pspe_CgoreauiAligned.out.bam --outFileNamePrefix 20220228_Meso_Pspe_10_TF_Pspe_CgoreauiAligned_deduped.bamProcessed.out.bam

crun.salmon quant -t /cm/shared/courses/dbarshis/barshislab/VRad/taxons/Pachyseris_speciosa/data_RNAseq/Pspe_Cgoreaui_hybridref.fasta --libType A -a 20220228_Meso_Pspe_10_TF_Pspe_CgoreauiAligned_deduped.bamProcessed.out.bam -o 20220228_Meso_Pspe_10_TF_Pspe_CgoreauiAligned_deduped_salmon_quant

STAR --runThreadN 32 --runMode inputAlignmentsFromBAM --bamRemoveDuplicatesType UniqueIdenticalNotMulti --inputBAMfile 20220228_Meso_Pspe_1_C_T2_Pspe_CgoreauiAligned.out.bam --outFileNamePrefix 20220228_Meso_Pspe_1_C_T2_Pspe_CgoreauiAligned_deduped.bamProcessed.out.bam

crun.salmon quant -t /cm/shared/courses/dbarshis/barshislab/VRad/taxons/Pachyseris_speciosa/data_RNAseq/Pspe_Cgoreaui_hybridref.fasta --libType A -a 20220228_Meso_Pspe_1_C_T2_Pspe_CgoreauiAligned_deduped.bamProcessed.out.bam -o 20220228_Meso_Pspe_1_C_T2_Pspe_CgoreauiAligned_deduped_salmon_quant

STAR --runThreadN 32 --runMode inputAlignmentsFromBAM --bamRemoveDuplicatesType UniqueIdenticalNotMulti --inputBAMfile 20220228_Meso_Pspe_1_C_T3_Pspe_CgoreauiAligned.out.bam --outFileNamePrefix 20220228_Meso_Pspe_1_C_T3_Pspe_CgoreauiAligned_deduped.bamProcessed.out.bam

crun.salmon quant -t /cm/shared/courses/dbarshis/barshislab/VRad/taxons/Pachyseris_speciosa/data_RNAseq/Pspe_Cgoreaui_hybridref.fasta --libType A -a 20220228_Meso_Pspe_1_C_T3_Pspe_CgoreauiAligned_deduped.bamProcessed.out.bam -o 20220228_Meso_Pspe_1_C_T3_Pspe_CgoreauiAligned_deduped_salmon_quant

STAR --runThreadN 32 --runMode inputAlignmentsFromBAM --bamRemoveDuplicatesType UniqueIdenticalNotMulti --inputBAMfile 20220228_Meso_Pspe_1_H_T2_Pspe_CgoreauiAligned.out.bam --outFileNamePrefix 20220228_Meso_Pspe_1_H_T2_Pspe_CgoreauiAligned_deduped.bamProcessed.out.bam

crun.salmon quant -t /cm/shared/courses/dbarshis/barshislab/VRad/taxons/Pachyseris_speciosa/data_RNAseq/Pspe_Cgoreaui_hybridref.fasta --libType A -a 20220228_Meso_Pspe_1_H_T2_Pspe_CgoreauiAligned_deduped.bamProcessed.out.bam -o 20220228_Meso_Pspe_1_H_T2_Pspe_CgoreauiAligned_deduped_salmon_quant

STAR --runThreadN 32 --runMode inputAlignmentsFromBAM --bamRemoveDuplicatesType UniqueIdenticalNotMulti --inputBAMfile 20220228_Meso_Pspe_1_L_T2_Pspe_CgoreauiAligned.out.bam --outFileNamePrefix 20220228_Meso_Pspe_1_L_T2_Pspe_CgoreauiAligned_deduped.bamProcessed.out.bam

crun.salmon quant -t /cm/shared/courses/dbarshis/barshislab/VRad/taxons/Pachyseris_speciosa/data_RNAseq/Pspe_Cgoreaui_hybridref.fasta --libType A -a 20220228_Meso_Pspe_1_L_T2_Pspe_CgoreauiAligned_deduped.bamProcessed.out.bam -o 20220228_Meso_Pspe_1_L_T2_Pspe_CgoreauiAligned_deduped_salmon_quant

STAR --runThreadN 32 --runMode inputAlignmentsFromBAM --bamRemoveDuplicatesType UniqueIdenticalNotMulti --inputBAMfile 20220228_Meso_Pspe_1_L_T3_Pspe_CgoreauiAligned.out.bam --outFileNamePrefix 20220228_Meso_Pspe_1_L_T3_Pspe_CgoreauiAligned_deduped.bamProcessed.out.bam

crun.salmon quant -t /cm/shared/courses/dbarshis/barshislab/VRad/taxons/Pachyseris_speciosa/data_RNAseq/Pspe_Cgoreaui_hybridref.fasta --libType A -a 20220228_Meso_Pspe_1_L_T3_Pspe_CgoreauiAligned_deduped.bamProcessed.out.bam -o 20220228_Meso_Pspe_1_L_T3_Pspe_CgoreauiAligned_deduped_salmon_quant

STAR --runThreadN 32 --runMode inputAlignmentsFromBAM --bamRemoveDuplicatesType UniqueIdenticalNotMulti --inputBAMfile 20220228_Meso_Pspe_1_M_T2_Pspe_CgoreauiAligned.out.bam --outFileNamePrefix 20220228_Meso_Pspe_1_M_T2_Pspe_CgoreauiAligned_deduped.bamProcessed.out.bam

crun.salmon quant -t /cm/shared/courses/dbarshis/barshislab/VRad/taxons/Pachyseris_speciosa/data_RNAseq/Pspe_Cgoreaui_hybridref.fasta --libType A -a 20220228_Meso_Pspe_1_M_T2_Pspe_CgoreauiAligned_deduped.bamProcessed.out.bam -o 20220228_Meso_Pspe_1_M_T2_Pspe_CgoreauiAligned_deduped_salmon_quant

STAR --runThreadN 32 --runMode inputAlignmentsFromBAM --bamRemoveDuplicatesType UniqueIdenticalNotMulti --inputBAMfile 20220228_Meso_Pspe_1_M_T3_Pspe_CgoreauiAligned.out.bam --outFileNamePrefix 20220228_Meso_Pspe_1_M_T3_Pspe_CgoreauiAligned_deduped.bamProcessed.out.bam

crun.salmon quant -t /cm/shared/courses/dbarshis/barshislab/VRad/taxons/Pachyseris_speciosa/data_RNAseq/Pspe_Cgoreaui_hybridref.fasta --libType A -a 20220228_Meso_Pspe_1_M_T3_Pspe_CgoreauiAligned_deduped.bamProcessed.out.bam -o 20220228_Meso_Pspe_1_M_T3_Pspe_CgoreauiAligned_deduped_salmon_quant

STAR --runThreadN 32 --runMode inputAlignmentsFromBAM --bamRemoveDuplicatesType UniqueIdenticalNotMulti --inputBAMfile 20220228_Meso_Pspe_1_T0_Pspe_CgoreauiAligned.out.bam --outFileNamePrefix 20220228_Meso_Pspe_1_T0_Pspe_CgoreauiAligned_deduped.bamProcessed.out.bam

crun.salmon quant -t /cm/shared/courses/dbarshis/barshislab/VRad/taxons/Pachyseris_speciosa/data_RNAseq/Pspe_Cgoreaui_hybridref.fasta --libType A -a 20220228_Meso_Pspe_1_T0_Pspe_CgoreauiAligned_deduped.bamProcessed.out.bam -o 20220228_Meso_Pspe_1_T0_Pspe_CgoreauiAligned_deduped_salmon_quant

STAR --runThreadN 32 --runMode inputAlignmentsFromBAM --bamRemoveDuplicatesType UniqueIdenticalNotMulti --inputBAMfile 20220228_Meso_Pspe_1_TF_Pspe_CgoreauiAligned.out.bam --outFileNamePrefix 20220228_Meso_Pspe_1_TF_Pspe_CgoreauiAligned_deduped.bamProcessed.out.bam

crun.salmon quant -t /cm/shared/courses/dbarshis/barshislab/VRad/taxons/Pachyseris_speciosa/data_RNAseq/Pspe_Cgoreaui_hybridref.fasta --libType A -a 20220228_Meso_Pspe_1_TF_Pspe_CgoreauiAligned_deduped.bamProcessed.out.bam -o 20220228_Meso_Pspe_1_TF_Pspe_CgoreauiAligned_deduped_salmon_quant

STAR --runThreadN 32 --runMode inputAlignmentsFromBAM --bamRemoveDuplicatesType UniqueIdenticalNotMulti --inputBAMfile 20220228_Meso_Pspe_3_C_T2_Pspe_CgoreauiAligned.out.bam --outFileNamePrefix 20220228_Meso_Pspe_3_C_T2_Pspe_CgoreauiAligned_deduped.bamProcessed.out.bam

crun.salmon quant -t /cm/shared/courses/dbarshis/barshislab/VRad/taxons/Pachyseris_speciosa/data_RNAseq/Pspe_Cgoreaui_hybridref.fasta --libType A -a 20220228_Meso_Pspe_3_C_T2_Pspe_CgoreauiAligned_deduped.bamProcessed.out.bam -o 20220228_Meso_Pspe_3_C_T2_Pspe_CgoreauiAligned_deduped_salmon_quant

STAR --runThreadN 32 --runMode inputAlignmentsFromBAM --bamRemoveDuplicatesType UniqueIdenticalNotMulti --inputBAMfile 20220228_Meso_Pspe_3_H_T2_Pspe_CgoreauiAligned.out.bam --outFileNamePrefix 20220228_Meso_Pspe_3_H_T2_Pspe_CgoreauiAligned_deduped.bamProcessed.out.bam

crun.salmon quant -t /cm/shared/courses/dbarshis/barshislab/VRad/taxons/Pachyseris_speciosa/data_RNAseq/Pspe_Cgoreaui_hybridref.fasta --libType A -a 20220228_Meso_Pspe_3_H_T2_Pspe_CgoreauiAligned_deduped.bamProcessed.out.bam -o 20220228_Meso_Pspe_3_H_T2_Pspe_CgoreauiAligned_deduped_salmon_quant

STAR --runThreadN 32 --runMode inputAlignmentsFromBAM --bamRemoveDuplicatesType UniqueIdenticalNotMulti --inputBAMfile 20220228_Meso_Pspe_3_L_T2_Pspe_CgoreauiAligned.out.bam --outFileNamePrefix 20220228_Meso_Pspe_3_L_T2_Pspe_CgoreauiAligned_deduped.bamProcessed.out.bam

crun.salmon quant -t /cm/shared/courses/dbarshis/barshislab/VRad/taxons/Pachyseris_speciosa/data_RNAseq/Pspe_Cgoreaui_hybridref.fasta --libType A -a 20220228_Meso_Pspe_3_L_T2_Pspe_CgoreauiAligned_deduped.bamProcessed.out.bam -o 20220228_Meso_Pspe_3_L_T2_Pspe_CgoreauiAligned_deduped_salmon_quant

STAR --runThreadN 32 --runMode inputAlignmentsFromBAM --bamRemoveDuplicatesType UniqueIdenticalNotMulti --inputBAMfile 20220228_Meso_Pspe_3_L_T3_Pspe_CgoreauiAligned.out.bam --outFileNamePrefix 20220228_Meso_Pspe_3_L_T3_Pspe_CgoreauiAligned_deduped.bamProcessed.out.bam

crun.salmon quant -t /cm/shared/courses/dbarshis/barshislab/VRad/taxons/Pachyseris_speciosa/data_RNAseq/Pspe_Cgoreaui_hybridref.fasta --libType A -a 20220228_Meso_Pspe_3_L_T3_Pspe_CgoreauiAligned_deduped.bamProcessed.out.bam -o 20220228_Meso_Pspe_3_L_T3_Pspe_CgoreauiAligned_deduped_salmon_quant

STAR --runThreadN 32 --runMode inputAlignmentsFromBAM --bamRemoveDuplicatesType UniqueIdenticalNotMulti --inputBAMfile 20220228_Meso_Pspe_3_M_T2_Pspe_CgoreauiAligned.out.bam --outFileNamePrefix 20220228_Meso_Pspe_3_M_T2_Pspe_CgoreauiAligned_deduped.bamProcessed.out.bam

crun.salmon quant -t /cm/shared/courses/dbarshis/barshislab/VRad/taxons/Pachyseris_speciosa/data_RNAseq/Pspe_Cgoreaui_hybridref.fasta --libType A -a 20220228_Meso_Pspe_3_M_T2_Pspe_CgoreauiAligned_deduped.bamProcessed.out.bam -o 20220228_Meso_Pspe_3_M_T2_Pspe_CgoreauiAligned_deduped_salmon_quant

STAR --runThreadN 32 --runMode inputAlignmentsFromBAM --bamRemoveDuplicatesType UniqueIdenticalNotMulti --inputBAMfile 20220228_Meso_Pspe_3_M_T3_Pspe_CgoreauiAligned.out.bam --outFileNamePrefix 20220228_Meso_Pspe_3_M_T3_Pspe_CgoreauiAligned_deduped.bamProcessed.out.bam

crun.salmon quant -t /cm/shared/courses/dbarshis/barshislab/VRad/taxons/Pachyseris_speciosa/data_RNAseq/Pspe_Cgoreaui_hybridref.fasta --libType A -a 20220228_Meso_Pspe_3_M_T3_Pspe_CgoreauiAligned_deduped.bamProcessed.out.bam -o 20220228_Meso_Pspe_3_M_T3_Pspe_CgoreauiAligned_deduped_salmon_quant

STAR --runThreadN 32 --runMode inputAlignmentsFromBAM --bamRemoveDuplicatesType UniqueIdenticalNotMulti --inputBAMfile 20220228_Meso_Pspe_3_T0_Pspe_CgoreauiAligned.out.bam --outFileNamePrefix 20220228_Meso_Pspe_3_T0_Pspe_CgoreauiAligned_deduped.bamProcessed.out.bam

crun.salmon quant -t /cm/shared/courses/dbarshis/barshislab/VRad/taxons/Pachyseris_speciosa/data_RNAseq/Pspe_Cgoreaui_hybridref.fasta --libType A -a 20220228_Meso_Pspe_3_T0_Pspe_CgoreauiAligned_deduped.bamProcessed.out.bam -o 20220228_Meso_Pspe_3_T0_Pspe_CgoreauiAligned_deduped_salmon_quant

STAR --runThreadN 32 --runMode inputAlignmentsFromBAM --bamRemoveDuplicatesType UniqueIdenticalNotMulti --inputBAMfile 20220228_Meso_Pspe_3_TF_Pspe_CgoreauiAligned.out.bam --outFileNamePrefix 20220228_Meso_Pspe_3_TF_Pspe_CgoreauiAligned_deduped.bamProcessed.out.bam

crun.salmon quant -t /cm/shared/courses/dbarshis/barshislab/VRad/taxons/Pachyseris_speciosa/data_RNAseq/Pspe_Cgoreaui_hybridref.fasta --libType A -a 20220228_Meso_Pspe_3_TF_Pspe_CgoreauiAligned_deduped.bamProcessed.out.bam -o 20220228_Meso_Pspe_3_TF_Pspe_CgoreauiAligned_deduped_salmon_quant

STAR --runThreadN 32 --runMode inputAlignmentsFromBAM --bamRemoveDuplicatesType UniqueIdenticalNotMulti --inputBAMfile 20220228_Meso_Pspe_6_C_T2_Pspe_CgoreauiAligned.out.bam --outFileNamePrefix 20220228_Meso_Pspe_6_C_T2_Pspe_CgoreauiAligned_deduped.bamProcessed.out.bam

crun.salmon quant -t /cm/shared/courses/dbarshis/barshislab/VRad/taxons/Pachyseris_speciosa/data_RNAseq/Pspe_Cgoreaui_hybridref.fasta --libType A -a 20220228_Meso_Pspe_6_C_T2_Pspe_CgoreauiAligned_deduped.bamProcessed.out.bam -o 20220228_Meso_Pspe_6_C_T2_Pspe_CgoreauiAligned_deduped_salmon_quant

STAR --runThreadN 32 --runMode inputAlignmentsFromBAM --bamRemoveDuplicatesType UniqueIdenticalNotMulti --inputBAMfile 20220228_Meso_Pspe_6_C_T3_Pspe_CgoreauiAligned.out.bam --outFileNamePrefix 20220228_Meso_Pspe_6_C_T3_Pspe_CgoreauiAligned_deduped.bamProcessed.out.bam

crun.salmon quant -t /cm/shared/courses/dbarshis/barshislab/VRad/taxons/Pachyseris_speciosa/data_RNAseq/Pspe_Cgoreaui_hybridref.fasta --libType A -a 20220228_Meso_Pspe_6_C_T3_Pspe_CgoreauiAligned_deduped.bamProcessed.out.bam -o 20220228_Meso_Pspe_6_C_T3_Pspe_CgoreauiAligned_deduped_salmon_quant

STAR --runThreadN 32 --runMode inputAlignmentsFromBAM --bamRemoveDuplicatesType UniqueIdenticalNotMulti --inputBAMfile 20220228_Meso_Pspe_6_H_T2_Pspe_CgoreauiAligned.out.bam --outFileNamePrefix 20220228_Meso_Pspe_6_H_T2_Pspe_CgoreauiAligned_deduped.bamProcessed.out.bam

crun.salmon quant -t /cm/shared/courses/dbarshis/barshislab/VRad/taxons/Pachyseris_speciosa/data_RNAseq/Pspe_Cgoreaui_hybridref.fasta --libType A -a 20220228_Meso_Pspe_6_H_T2_Pspe_CgoreauiAligned_deduped.bamProcessed.out.bam -o 20220228_Meso_Pspe_6_H_T2_Pspe_CgoreauiAligned_deduped_salmon_quant

STAR --runThreadN 32 --runMode inputAlignmentsFromBAM --bamRemoveDuplicatesType UniqueIdenticalNotMulti --inputBAMfile 20220228_Meso_Pspe_6_L_T2_Pspe_CgoreauiAligned.out.bam --outFileNamePrefix 20220228_Meso_Pspe_6_L_T2_Pspe_CgoreauiAligned_deduped.bamProcessed.out.bam

crun.salmon quant -t /cm/shared/courses/dbarshis/barshislab/VRad/taxons/Pachyseris_speciosa/data_RNAseq/Pspe_Cgoreaui_hybridref.fasta --libType A -a 20220228_Meso_Pspe_6_L_T2_Pspe_CgoreauiAligned_deduped.bamProcessed.out.bam -o 20220228_Meso_Pspe_6_L_T2_Pspe_CgoreauiAligned_deduped_salmon_quant

STAR --runThreadN 32 --runMode inputAlignmentsFromBAM --bamRemoveDuplicatesType UniqueIdenticalNotMulti --inputBAMfile 20220228_Meso_Pspe_6_L_T3_Pspe_CgoreauiAligned.out.bam --outFileNamePrefix 20220228_Meso_Pspe_6_L_T3_Pspe_CgoreauiAligned_deduped.bamProcessed.out.bam

crun.salmon quant -t /cm/shared/courses/dbarshis/barshislab/VRad/taxons/Pachyseris_speciosa/data_RNAseq/Pspe_Cgoreaui_hybridref.fasta --libType A -a 20220228_Meso_Pspe_6_L_T3_Pspe_CgoreauiAligned_deduped.bamProcessed.out.bam -o 20220228_Meso_Pspe_6_L_T3_Pspe_CgoreauiAligned_deduped_salmon_quant

STAR --runThreadN 32 --runMode inputAlignmentsFromBAM --bamRemoveDuplicatesType UniqueIdenticalNotMulti --inputBAMfile 20220228_Meso_Pspe_6_M_T2_Pspe_CgoreauiAligned.out.bam --outFileNamePrefix 20220228_Meso_Pspe_6_M_T2_Pspe_CgoreauiAligned_deduped.bamProcessed.out.bam

crun.salmon quant -t /cm/shared/courses/dbarshis/barshislab/VRad/taxons/Pachyseris_speciosa/data_RNAseq/Pspe_Cgoreaui_hybridref.fasta --libType A -a 20220228_Meso_Pspe_6_M_T2_Pspe_CgoreauiAligned_deduped.bamProcessed.out.bam -o 20220228_Meso_Pspe_6_M_T2_Pspe_CgoreauiAligned_deduped_salmon_quant

STAR --runThreadN 32 --runMode inputAlignmentsFromBAM --bamRemoveDuplicatesType UniqueIdenticalNotMulti --inputBAMfile 20220228_Meso_Pspe_6_M_T3_Pspe_CgoreauiAligned.out.bam --outFileNamePrefix 20220228_Meso_Pspe_6_M_T3_Pspe_CgoreauiAligned_deduped.bamProcessed.out.bam

crun.salmon quant -t /cm/shared/courses/dbarshis/barshislab/VRad/taxons/Pachyseris_speciosa/data_RNAseq/Pspe_Cgoreaui_hybridref.fasta --libType A -a 20220228_Meso_Pspe_6_M_T3_Pspe_CgoreauiAligned_deduped.bamProcessed.out.bam -o 20220228_Meso_Pspe_6_M_T3_Pspe_CgoreauiAligned_deduped_salmon_quant

STAR --runThreadN 32 --runMode inputAlignmentsFromBAM --bamRemoveDuplicatesType UniqueIdenticalNotMulti --inputBAMfile 20220228_Meso_Pspe_6_T0_Pspe_CgoreauiAligned.out.bam --outFileNamePrefix 20220228_Meso_Pspe_6_T0_Pspe_CgoreauiAligned_deduped.bamProcessed.out.bam

crun.salmon quant -t /cm/shared/courses/dbarshis/barshislab/VRad/taxons/Pachyseris_speciosa/data_RNAseq/Pspe_Cgoreaui_hybridref.fasta --libType A -a 20220228_Meso_Pspe_6_T0_Pspe_CgoreauiAligned_deduped.bamProcessed.out.bam -o 20220228_Meso_Pspe_6_T0_Pspe_CgoreauiAligned_deduped_salmon_quant

STAR --runThreadN 32 --runMode inputAlignmentsFromBAM --bamRemoveDuplicatesType UniqueIdenticalNotMulti --inputBAMfile 20220228_Meso_Pspe_6_TF_Pspe_CgoreauiAligned.out.bam --outFileNamePrefix 20220228_Meso_Pspe_6_TF_Pspe_CgoreauiAligned_deduped.bamProcessed.out.bam

crun.salmon quant -t /cm/shared/courses/dbarshis/barshislab/VRad/taxons/Pachyseris_speciosa/data_RNAseq/Pspe_Cgoreaui_hybridref.fasta --libType A -a 20220228_Meso_Pspe_6_TF_Pspe_CgoreauiAligned_deduped.bamProcessed.out.bam -o 20220228_Meso_Pspe_6_TF_Pspe_CgoreauiAligned_deduped_salmon_quant

STAR --runThreadN 32 --runMode inputAlignmentsFromBAM --bamRemoveDuplicatesType UniqueIdenticalNotMulti --inputBAMfile 20220228_Meso_Pspe_8_C_T2_Pspe_CgoreauiAligned.out.bam --outFileNamePrefix 20220228_Meso_Pspe_8_C_T2_Pspe_CgoreauiAligned_deduped.bamProcessed.out.bam

crun.salmon quant -t /cm/shared/courses/dbarshis/barshislab/VRad/taxons/Pachyseris_speciosa/data_RNAseq/Pspe_Cgoreaui_hybridref.fasta --libType A -a 20220228_Meso_Pspe_8_C_T2_Pspe_CgoreauiAligned_deduped.bamProcessed.out.bam -o 20220228_Meso_Pspe_8_C_T2_Pspe_CgoreauiAligned_deduped_salmon_quant

STAR --runThreadN 32 --runMode inputAlignmentsFromBAM --bamRemoveDuplicatesType UniqueIdenticalNotMulti --inputBAMfile 20220228_Meso_Pspe_8_C_T3_Pspe_CgoreauiAligned.out.bam --outFileNamePrefix 20220228_Meso_Pspe_8_C_T3_Pspe_CgoreauiAligned_deduped.bamProcessed.out.bam

crun.salmon quant -t /cm/shared/courses/dbarshis/barshislab/VRad/taxons/Pachyseris_speciosa/data_RNAseq/Pspe_Cgoreaui_hybridref.fasta --libType A -a 20220228_Meso_Pspe_8_C_T3_Pspe_CgoreauiAligned_deduped.bamProcessed.out.bam -o 20220228_Meso_Pspe_8_C_T3_Pspe_CgoreauiAligned_deduped_salmon_quant

STAR --runThreadN 32 --runMode inputAlignmentsFromBAM --bamRemoveDuplicatesType UniqueIdenticalNotMulti --inputBAMfile 20220228_Meso_Pspe_8_H_T2_Pspe_CgoreauiAligned.out.bam --outFileNamePrefix 20220228_Meso_Pspe_8_H_T2_Pspe_CgoreauiAligned_deduped.bamProcessed.out.bam

crun.salmon quant -t /cm/shared/courses/dbarshis/barshislab/VRad/taxons/Pachyseris_speciosa/data_RNAseq/Pspe_Cgoreaui_hybridref.fasta --libType A -a 20220228_Meso_Pspe_8_H_T2_Pspe_CgoreauiAligned_deduped.bamProcessed.out.bam -o 20220228_Meso_Pspe_8_H_T2_Pspe_CgoreauiAligned_deduped_salmon_quant

STAR --runThreadN 32 --runMode inputAlignmentsFromBAM --bamRemoveDuplicatesType UniqueIdenticalNotMulti --inputBAMfile 20220228_Meso_Pspe_8_L_T2_Pspe_CgoreauiAligned.out.bam --outFileNamePrefix 20220228_Meso_Pspe_8_L_T2_Pspe_CgoreauiAligned_deduped.bamProcessed.out.bam

crun.salmon quant -t /cm/shared/courses/dbarshis/barshislab/VRad/taxons/Pachyseris_speciosa/data_RNAseq/Pspe_Cgoreaui_hybridref.fasta --libType A -a 20220228_Meso_Pspe_8_L_T2_Pspe_CgoreauiAligned_deduped.bamProcessed.out.bam -o 20220228_Meso_Pspe_8_L_T2_Pspe_CgoreauiAligned_deduped_salmon_quant

STAR --runThreadN 32 --runMode inputAlignmentsFromBAM --bamRemoveDuplicatesType UniqueIdenticalNotMulti --inputBAMfile 20220228_Meso_Pspe_8_L_T3_Pspe_CgoreauiAligned.out.bam --outFileNamePrefix 20220228_Meso_Pspe_8_L_T3_Pspe_CgoreauiAligned_deduped.bamProcessed.out.bam

crun.salmon quant -t /cm/shared/courses/dbarshis/barshislab/VRad/taxons/Pachyseris_speciosa/data_RNAseq/Pspe_Cgoreaui_hybridref.fasta --libType A -a 20220228_Meso_Pspe_8_L_T3_Pspe_CgoreauiAligned_deduped.bamProcessed.out.bam -o 20220228_Meso_Pspe_8_L_T3_Pspe_CgoreauiAligned_deduped_salmon_quant

STAR --runThreadN 32 --runMode inputAlignmentsFromBAM --bamRemoveDuplicatesType UniqueIdenticalNotMulti --inputBAMfile 20220228_Meso_Pspe_8_M_T2_Pspe_CgoreauiAligned.out.bam --outFileNamePrefix 20220228_Meso_Pspe_8_M_T2_Pspe_CgoreauiAligned_deduped.bamProcessed.out.bam

crun.salmon quant -t /cm/shared/courses/dbarshis/barshislab/VRad/taxons/Pachyseris_speciosa/data_RNAseq/Pspe_Cgoreaui_hybridref.fasta --libType A -a 20220228_Meso_Pspe_8_M_T2_Pspe_CgoreauiAligned_deduped.bamProcessed.out.bam -o 20220228_Meso_Pspe_8_M_T2_Pspe_CgoreauiAligned_deduped_salmon_quant

STAR --runThreadN 32 --runMode inputAlignmentsFromBAM --bamRemoveDuplicatesType UniqueIdenticalNotMulti --inputBAMfile 20220228_Meso_Pspe_8_T0_Pspe_CgoreauiAligned.out.bam --outFileNamePrefix 20220228_Meso_Pspe_8_T0_Pspe_CgoreauiAligned.out_deduped.bamProcessed.out.bam

crun.salmon quant -t /cm/shared/courses/dbarshis/barshislab/VRad/taxons/Pachyseris_speciosa/data_RNAseq/Pspe_Cgoreaui_hybridref.fasta --libType A -a 20220228_Meso_Pspe_8_T0_Pspe_CgoreauiAligned.out_deduped.bamProcessed.out.bam -o 20220228_Meso_Pspe_8_T0_Pspe_CgoreauiAligned.out_deduped_salmon_quant

STAR --runThreadN 32 --runMode inputAlignmentsFromBAM --bamRemoveDuplicatesType UniqueIdenticalNotMulti --inputBAMfile 20220228_Meso_Pspe_8_TF_Pspe_CgoreauiAligned.out.bam --outFileNamePrefix 20220228_Meso_Pspe_8_TF_Pspe_CgoreauiAligned.out_deduped.bamProcessed.out.bam

crun.salmon quant -t /cm/shared/courses/dbarshis/barshislab/VRad/taxons/Pachyseris_speciosa/data_RNAseq/Pspe_Cgoreaui_hybridref.fasta --libType A -a 20220228_Meso_Pspe_8_TF_Pspe_CgoreauiAligned.out_deduped.bamProcessed.out.bam -o 20220228_Meso_Pspe_8_TF_Pspe_CgoreauiAligned
```

```
sbatch STAR-dedup_salmon-quant.sh
```

CONTINUE HERE

--------------------------------------------------------------------------------------------

?

These transcript expression estimates, often referred to as 'pseudocounts', can be converted for use with DGE tools like DESeq2 (using tximport) or the estimates can be used directly for isoform-level differential expression using a tool like Sleuth.

https://github.com/hbctraining/Intro-to-rnaseq-hpc-O2/blob/0feca5559bcbde27cbc7634085f07b3624a36f2f/lessons/08_salmon.md

What is Salmon?
Salmon is based on the philosophy of lightweight algorithms, which use the reference transcriptome (in FASTA format) and raw sequencing reads (in FASTQ format) as input, but do not align the full reads. These tools perform both mapping and quantification. Unlike most lightweight and standard alignment/quantification tools, Salmon utilizes sample-specific bias models for transcriptome-wide abundance estimation. Sample-specific bias models are helpful when needing to account for known biases present in RNA-Seq data including:

- GC bias

- positional coverage biases

- sequence biases at 5' and 3' ends of the fragments

- fragment length distribution

- strand-specific methods


If not accounted for, these biases can lead to unacceptable false positive rates in differential expression studies [2]. The Salmon algorithm can learn these sample-specific biases and account for them in the transcript abundance estimates. Salmon is extremely fast at "mapping" reads to the transcriptome and often more accurate than standard approaches [2].

--------------------------------------------------------------------------------------------

# Move on to DESeq2 in R

Import the salmon quant counts into DESeq2 with tximport.

---------------------------------------------------------------------------------------------

